shopfloor_reception/actions/search.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.addons.component.core import Component
+
+
+class SearchAction(Component):
+
+    _inherit = "shopfloor.search.action"
+
+    def _get_origin_from_barcode(self, barcode, **kwargs):
+        if kwargs.get("reception_return"):
+            carriers = self.env["delivery.carrier"].search([])
+            res = carriers._get_origin_from_barcode(barcode)
+            if res:
+                return res
+        return super()._get_origin_from_barcode(barcode)
```

Comment:
Could we just move this into the reception scenario, it is only used there. 
This would then simplify the code. 
Also it is not needed to search all carriers. If we merge https://github.com/OCA/delivery-carrier/pull/767
```suggestion
    def _get_origin_from_barcode(self, barcode, **kwargs):
        if kwargs.get("reception_return"):
            res = self.env["delivery.carrier"]._get_origin_from_barcode(barcode)
            if res:
                return res
        return super()._get_origin_from_barcode(barcode)
```

Reply:
I agree. Updated